Continuous Integration with Travis CI

GochoMugo opened the pull request:
[Travis CI](https://travis-ci.org) offers free CI services for Open Source projects. It relies on [build configuration](http://docs.travis-ci.com/user/build-configuration/) specified in a `.travis.yml` file in the root of the repository. See [Getting Started with Travis CI](http://docs.travis-ci.com/user/getting-started/).

Some minor changes to the source files have made, notably:
- Reading Port number from environment returns it as a string. We require to convert it to an integer when calling `MySQLdb.Connect` &mdash; 9a97d5a
- A new Makefile target, `reqs`, allows manual installation of requirements such as `MySQL-python`. The requirements are stated per line in the `requirements.txt` file &mdash; 4850793
- We only ignore the `*.yml` files in the `tests/` directory.

After merge, you may want to include a [build status image](http://docs.travis-ci.com/user/status-images/) in the `README.md`

GochoMugo committed:
Introducing Continuous Integration

CI (continuous integration) offered by Travis CI.

.travis.yml offers build configurations. The tests will be run in
two matrixes. One for MySQL and the other for SQLite3.

.gitignore made .travis.yml be ingored by git. Have it only ingore
.yml files in tests/, specifically the config.yml that holds
test configuration for developers.

For more information see:

- http://docs.travis-ci.com/user/build-configuration/
- http://docs.travis-ci.com/user/environment-variables/
- http://docs.travis-ci.com/user/database-setup/

GochoMugo committed:
Install Requirements before test

We require to install requirements before running the tests. This will
be achieved by the `reqs` target in the makefile: `make reqs`.
The CI server will invoke this command before test.

Developers may also install the requirements on their computer using
`make reqs`.

GochoMugo committed:
Error installing requirements

This error occurs while installing:

  error: /usr/local/lib/python2.7/dist-packages/_mysql.so: Permission denied

Trying to give Travis CI some sudo powers as demonstrated here:
http://docs.travis-ci.com/user/installing-dependencies/

Also the option, `--use-mirrors`, is deprecated. Ignoring this option
for now. May consider later.

GochoMugo committed:
Port number for MySQLdb must be an integer

Reading the port from the environment returns it as a string.
We require to convert to an integer before passing to
`MySQLdb.Connect` method.

GochoMugo committed:
Create Test Database before tests

Travis CI will create a MySQL database for running the tests in.

Refer to http://docs.travis-ci.com/user/database-setup/#MySQL for
more information

firstprayer commented:
Can you add some introduction in the README.md about this travis integration and how to set it up?

GochoMugo committed:
Instructions for running tests

Instructions for running tests either locally or from Travis CI have
been updated and added respectively.
